packages/surveys/src/components/general/LanguageSwitch.tsx

```diff
@@ -0,0 +1,85 @@
+import { useRef, useState } from "react";
+
+import { getLanguageLabel } from "@formbricks/lib/i18n/utils";
+import { useClickOutside } from "@formbricks/lib/utils/hooks/useClickOutside";
+import { TSurveyLanguage } from "@formbricks/types/surveys";
+
+interface LanguageSwitchProps {
+  selectedLanguageCode: string;
+  surveyLanguages: TSurveyLanguage[];
+  setSelectedLanguageCode: (languageCode: string) => void;
+  setFirstRender?: (firstRender: boolean) => void;
+}
+export const LanguageSwitch = ({
+  surveyLanguages,
+  selectedLanguageCode,
+  setSelectedLanguageCode,
+  setFirstRender,
+}: LanguageSwitchProps) => {
+  const [showLanguageDropdown, setShowLanguageDropdown] = useState(false);
+  const toggleDropdown = () => setShowLanguageDropdown((prev) => !prev);
+  const languageDropdownRef = useRef(null);
+  const defaultLangauge = surveyLanguages.find((surveyLanguage) => {
```

Comment:
Since we always use the language code, can we also extract the language code here directly?
`const defaultLanguageCode = ...language.code;`